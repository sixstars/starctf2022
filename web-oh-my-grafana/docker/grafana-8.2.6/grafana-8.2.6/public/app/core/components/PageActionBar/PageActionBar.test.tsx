import React from 'react';
import { shallow } from 'enzyme';
import PageActionBar, { Props } from './PageActionBar';

const setup = (propOverrides?: object) => {
  const props: Props = {
    searchQuery: '',
    setSearchQuery: jest.fn(),
    target: '_blank',
    linkButton: { href: 'some/url', title: 'test' },
  };

  Object.assign(props, propOverrides);

  return shallow(<PageActionBar {...props} />);
};

describe('Render', () => {
  it('should render component', () => {
    const wrapper = setup();

    expect(wrapper).toMatchSnapshot();
  });
});
