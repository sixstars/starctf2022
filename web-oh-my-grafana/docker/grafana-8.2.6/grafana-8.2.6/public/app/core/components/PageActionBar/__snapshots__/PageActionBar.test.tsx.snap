// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render should render component 1`] = `
<div
  className="page-action-bar"
>
  <div
    className="gf-form gf-form--grow"
  >
    <FilterInput
      onChange={[MockFunction]}
      placeholder="Search by name or type"
      value=""
    />
  </div>
  <LinkButton
    href="some/url"
    target="_blank"
  >
    test
  </LinkButton>
</div>
`;
